### PR TITLE
[Chore] Update fork

### PR DESCRIPTION
**Pull request by stevensJourney**
# Overview

This updates the fork from upstream.

# Testing

Verified that the demo loads the SQLite core extension correctly.
image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -148,9 +153,9 @@ deps/$(SQLITE_VERSION)/sqlite3.h deps/$(SQLITE_VERSION)/sqlite3.c:
 
 deps/$(EXTENSION_FUNCTIONS): cache/$(EXTENSION_FUNCTIONS)
 	mkdir -p deps
-	openssl dgst -sha3-256 -r cache/$(EXTENSION_FUNCTIONS) | sed -e 's/\s.*//' > deps/sha3
-	echo $(EXTENSION_FUNCTIONS_SHA3) | cmp deps/sha3
-	rm -rf deps/sha3 $@
+	# openssl dgst -sha256 -r cache/$(EXTENSION_FUNCTIONS) | sed -e 's/\s.*//' > deps/sha3
```
Comment (stevensJourney) [inline Makefile:156]: TODO: Need to remove this comment. This breaks on MacOS

Comment (stevensJourney) [reply Makefile]: Realized this was due to me using an older version of OpenSSL. I updated the README. Also improved this for MacOS versus Linux.

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # dependencies
+# TODO this should be 3.46.0, but there are build errors
 SQLITE_VERSION = version-3.44.0
```
Comment (stevensJourney) [inline Makefile:3]: I'm not sure why this fails for `3.46.0`. It seems like this error might be coming from our rust core
image

File: src/libvfs.c [hunk 3]
```diff
@@ -1,125 +1,231 @@
...
+#undef METHOD
+
+  vfs->methodMask = methodMask;
+  vfs->asyncMask = asyncMask;
+
+  *ppVfs = vfs;
+  return sqlite3_vfs_register(&vfs->base, makeDefault);
 }
 
 int setup_powersync()
```
Comment (rkistner) [inline src/libvfs.c:231]: Not directly related to this PR, but can the powersync setup be moved to main.c? libvfs.c is a weird place for it

Comment (stevensJourney) [reply src/libvfs.c:231]: Agreed. Will update.

**Comment by stevensJourney** [comment]
Superseded by https://github.com/powersync-ja/wa-sqlite/pull/19

**Pull request closed without merging**